friends/src/Components/logIn.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { connect } from 'react-redux';
+import { login } from '../actions';
+
+import Loader from 'react-loader-spinner'
+
+
+class Login extends React.Component {
+    state = {
+        credentials: {
+            username: '',
+            password: ''
+        }
+    };
+
+    handleChanges = e => {
+        this.setState({
+            credentials: {
+                ...this.state.credentials,
+                [e.target.name]: e.target.value
+            }
+        });
+    };
+
+    login = e => {
+        e.preventDefault();
+        this.props.login(this.state.credentials).then(() => {
+            this.props.history.push('/friends');
+        });
+    };
+    render() {
+        return (
+            <div className='login'>
+                <h2>Log In</h2>
+                <form onSubmit={this.login}>
+                    <input
+                        type='text'
+                        name='username'
+                        placeholder='Username'
+                        value={this.state.credentials.username}
+                        onChange={this.handleChanges}
+                    />
+                    <input
+                        type='text'
+                        name='password'
+                        placeholder='Password'
+                        value={this.state.credentials.password}
+                        onChange={this.handleChanges}
+                    />
+                    <button>
+                        {this.props.isLoggingIn ? (
+                            <Loader type="ThreeDots" color="#1f2a38" height="12" width="26" />
+                        ) : (
+                                'Log in'
+                            )}
+                    </button>
+                </form>
+            </div>
+        )
+    }
+}
+
+const mapStateToProps = state => ({
+    isLoggingIn: state.isLoggingIn
+})
+
```

Comment:
When you are using `combineReducer` (which is usually when you have more than one reducer), then you'll have to specify which reducer you are getting properties from in the `mapStateToProps` function. Here, it would be:
```
const mapStateToProps = state => ({
    isLoggingIn: state.loginReducer.isLoggingIn
})
```

Adding `loginReducer` as shown above will allow the loading icon to display on your Log In button.